pandas/core/dtypes/common.py

```diff
@@ -1668,7 +1668,7 @@ def infer_dtype_from_object(dtype) -> type:
         pass
 
     if is_extension_array_dtype(dtype):
-        return dtype.type
+        return type(dtype)
```

Comment:
I don't think this should change. According to the docstring, this function should return a numpy-style dtype (even from an ExtensionDtype

Reply:
ok noted.